fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/FedoraToOCFLObjectIndexUtilImplTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl.impl;
+
+import org.fcrepo.kernel.api.operations.CreateResourceOperation;
+import org.fcrepo.kernel.api.operations.NonRdfSourceOperation;
+import org.fcrepo.kernel.api.operations.RdfSourceOperation;
+import org.fcrepo.persistence.ocfl.api.FedoraOCFLMappingNotFoundException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.ByteArrayInputStream;
+
+import static org.fcrepo.kernel.api.operations.ResourceOperationType.CREATE;
+import static org.fcrepo.persistence.ocfl.impl.OCFLPersistentStorageUtils.createRepository;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+import static org.springframework.test.util.ReflectionTestUtils.setField;
+
+/**
+ * @author dbernstein
+ * @since 6.0.0
+ */
+public class FedoraToOCFLObjectIndexUtilImplTest {
+
+    private static final String RESOURCE_ID_1 = "info:fedora/resource";
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
```

Comment:
Unit tests should only write test artifacts to the Maven module's `target` directory.

Reply:
I can see some value in this, but `TemporaryFolder` is pretty standard junit and does guarantee a cleaner state between runs without having to perform a `maven clean`

Reply:
The question is, do we align more closely with Maven or with Junit? 
Maybe because I do all of my builds on the commandline, I align with Maven, and am adverse to the build leaving files/directories on my local machine after `mvn clean`.

Reply:
`TemporaryFolder` doesn't leave files on disk. But, you can have it both ways.

```java
@Rule
public TemporaryFolder tempFolder = new TemporaryFolder(new File("target/junit-temp"));
```

(Not sure if it will automatically handle creating the parent directory if it doesn't exist.)

Reply:
On the flip side, I do like to have test files left on-disk (`target`) in case I want to inspect them after a build... but want my system to be clean after running `mvn clean`.

Reply:
Fair enough! That is sometimes useful.

Reply:
I guess I view maven as the build tool and junit as the testing framework, with the hope that if something works in junit (particularly unit tests) it should work in maven or other test environments. I'm biased though since I usually run individual tests via an IDE, and mainly rely on maven when running the full suite. Usually in order to see what state was produced I end up setting break points, but its just a different workflow. 

I don't think it makes too much difference one way or the other for this particular case, so I'm okay with whatever we decide on here. It would be good to know if the project standard is to use `target` though. There are only a few places using TemporaryFolders at the moment, so maybe that is the case.

Reply:
Historically, Fedora has left a clean system after `mvn clean`. 
Agreed, if we want to change, we can... but it should be a decision.